File: src/main/java/org/example/libraryapp/errorHandler/BookNotFoundException.java
package org.example.libraryapp.errorHandler;

import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public class BookNotFoundException extends BaseException {

    final Long id;

    public BookNotFoundException(Long id) {
        super(HttpStatus.NOT_FOUND,
                new ApiError(404, "Not found", "Book with id " + id + " not found"));
        this.id = id;
    }
}

File: src/main/kotlin/org/example/libraryapp/errorHandler/BookNotFoundException.kt
package org.example.libraryapp.errorHandler

import org.springframework.http.HttpStatus

class BookNotFoundException(userId: Long) : BaseException(
    httpStatus = HttpStatus.NOT_FOUND,
    apiError =  ApiError(
        404,
        "NOT_FOUND",
        "Book with id $userId not found",
        )
)